### PR TITLE
[fix] update MAIN fetcher for new OSF filenames

### PR DESCRIPTION
**Pull request by emdupre**
Updates the fetcher and hashes to grab the OSF filenames with updated naming conventions !

### CODE REVIEW

**Comment by KamalakerDadi** [comment]
Merged. Thanks a lot!

File: nilearn/datasets/data/MAIN_osf.csv [hunk 1]
```diff
@@ -1,84 +1,84 @@
...
-w23s6,sub-pixar148,nha7y
-h9fsy,sub-pixar149,jebsq
-kw4ut,sub-pixar150,3jtwu
-dbw8y,sub-pixar151,gwns9
-rdchv,sub-pixar152,mhjrk
-sfkuh,sub-pixar153,btpv9
-7xzdq,sub-pixar154,yctrz
-b95rn,sub-pixar155,ut6nj
+participant_id,key_r,key_b
+sub-pixar001,5c8ff37da743a90018606df1,5c8ff37c2286e80019c3c102
```
Comment (KamalakerDadi) [inline nilearn/datasets/data/MAIN_osf.csv:2]: Do you know if we can automatically grab these keys ?

Comment (emdupre) [reply nilearn/datasets/data/MAIN_osf.csv:2]: Yes --  @rmarkello showed me how to use the OSF API :)

It looked like this:
```python
url = 'https://api.osf.io/v2/nodes/5hju4/files/osfstorage/?filter[name][contains]=confounds_regressors.tsv'
guids = []
while True:
    resp = requests.get(url)
    resp.raise_for_status()
    data = json.loads(resp.content)
    for i in data['data']:
        sub = re.search(r'sub-(\S+)_task', i['attributes']['name']).group(1)
        guids.append((sub, i['id']))

    url = data['links']['next']
    if url is None:
        break

```

Comment (KamalakerDadi) [reply nilearn/datasets/data/MAIN_osf.csv:2]: Sounds great. Thanks a lot!

**Pull request merged**